core/src/main/java/com/linecorp/armeria/server/file/HttpFileService.java

```diff
@@ -249,7 +249,16 @@ private Entry getEntry(String path, @Nullable String contentEncoding) {
             return config.vfs().get(path, contentEncoding);
         }
 
-        final CachedEntry entry = cache.get(new PathAndEncoding(path, contentEncoding));
+        final PathAndEncoding pathAndEncoding = new PathAndEncoding(path, contentEncoding);
+        CachedEntry entry = cache.getIfPresent(pathAndEncoding);
+        if (entry == null) {
+            entry = cache.get(pathAndEncoding);
+        } else {
+            if (config.vfs().get(path, contentEncoding).lastModifiedMillis() > entry.lastModifiedMillis()) {
```

Comment:
Not sure which is better but perhaps `!=` is OK here?

Reply:
yes, think all of the `cached file in entry` should from` config.vfs().get()` so there should be no possible that `config.vfs().get(path, contentEncoding).lastModifiedMillis() < entry.lastModifiedMillis()` right?